File: src/7/7.py

class Solution:
    def reverse1(self, x):
        res = 0
        while x!= 0:
            num = x%10
            x = x//10

    def reverse(self, x):
        res = 0
        flag = 1 if x > 0 else -1
        abs_x = abs(x)
        x_str = str(abs_x)
        reverse_x_str = x_str[::-1]
        reverse_x_int = int(reverse_x_str) * flag
        if -2**31 <= reverse_x_int <= 2**31-1:
            return reverse_x_int
        else:
            return 0


if __name__ == '__main__':
    sln = Solution()
    print(sln.reverse(-321))

File: src/7/7.cpp
#include <iostream>
#include <stack>
#include <vector>
#include <limits.h>

class Solution
{
public:
    int reverse(int x)
    {
	int temp = x;
	int res = 0;

	while (temp != 0)
	{
	    int num = temp % 10;
	    temp = temp/10;

	    if (res > INT_MAX/10 || res == INT_MAX/10 && num > 7)
		return 0;
	
	    if (res < INT_MIN/10 || res == INT_MAX/10 && num < -8)
		return 0;

	    res = res * 10 + num;
	}

	return res;
    }
};


int main()
{
    Solution sln;
    int res = sln.reverse(-3215);
    std::cout << "res : " << res << std::endl;

    return 0;
}
